app/models/comment.rb

```diff
@@ -25,4 +27,16 @@ def self.attributes
       'updated_at'
     ]
   end
+
+  # All of the users who should be notified when a comment is created
+  def listeners
+    users_to_notify = []
+    users_to_notify += self.proposal.currently_awaiting_approvers
+    users_to_notify += self.proposal.approvals.approved.map(&:user)
+    users_to_notify += self.proposal.observers
+    users_to_notify << self.proposal.requester
```

Comment:
`Set` makes sense. I'm not sure there's a good way to use `Proposal#users` here. We could try to _subtract_ the approvers that shouldn't be notified, but then we run the risk of someone appearing twice in the chain and being removed.